### ISSUE TITLE
Automate New Site Procedure

### ISSUE THREAD

**Issue by dougmet** [ONSdigital/sdg-indicators#2984]
How much of the current new-site walkthrough can be scripted to make the process less painful?

**Comment by SavvasStephanides** [comment]
This was the result of a lengthy discusion with @dougmet. There are several steps in the new site procedure that can be automated thus requiring less "human effort". 

## Examples of things that are simple to automate

* `ssh` key generation using `ssh-keygen`.

* Compressing files using `tar`

* Updating files (for example updating key codes in  `.sh` scripts

## Examples of things that might be trickier or not possible to automate 

* Logging in to GitHub

* Creation of repositories

* Logging in to CircleCI / TravisCI with GitHub.

* Changing settings to repositories in CI systems.

⚠️ *Some GitHub processes might be possible by using the [GitHub API](https://developer.github.com/v3/). Some investigation required* 

## Further considerations

### Docker

Some of the issues that our of our control are system specific, for example how/if `ssh-keygen` works on Windows, or which version of Python each user has downloaded. This can be resolved by introducing [Docker](https://www.docker.com/). Simply put, Docker creates a container with its own operating system and installations of software of our choosing. This means that all users will have the same OS/software and ensure that the process will work everywhere. Docker is free and easy to set up and the containers are pretty lightweight.

**Comment by dougmet** [comment]
Regarding logging into GitHub / Travis. The `travis` gem does some of this for you. And you can log in with a token instead of a password, so maybe there's something down that route. Perhaps the user can generate one token for the process and then disable it when they're done.

Also could think about coming from the other direction. Editing the scripts to play nicely with automation. More use of variables and less code duplication between scripts.

**Comment by SavvasStephanides** [comment]
I've started this project to test a few things out: https://github.com/SavvasStephanides/sdg-autostarter

So far I've managed to create Python scripts (see `content` folder) for logging in to GitHub and forking the two repositories. It seems that there's no option to create organizations (eg. `sustainabledevelopment-xyz`) using the GitHub API but I'll check further.

Also I'm holding any investigation on CI systems for now until the meeting next week.

Let me know what you think so far. I've provided instructions on how to run this using Docker.
